parsers/GB_NIR.py

```diff
@@ -0,0 +1,166 @@
+from datetime import datetime, timedelta
+from logging import Logger, getLogger
+from typing import Optional
+
+from dateutil import tz
+from requests import ConnectionError, Response, Session
+
+from parsers.lib.config import refetch_frequency
+
+TZ = "Europe/Dublin"
+DATA_URL = "https://www.smartgriddashboard.com/DashboardService.svc/data"
+
+
+def _get_previous_quarter(time: datetime):
+    last_quarter_minute = 15 * (time.minute // 15)
+    return time.replace(minute=last_quarter_minute)
+
+
+def _parse_effective_time(time_str: str):
+    naive_dt = datetime.strptime(time_str, "%d-%b-%Y %H:%M:%S")
+    return naive_dt.replace(tzinfo=tz.gettz(name=TZ))
+
+
+def _fetch_json_data(
+    region: str,
+    dataset: str,
+    session: Session = Session(),
+    target_datetime: Optional[datetime] = None,
+) -> dict:
+    try:
+        if not target_datetime:
+            # get the previous datapoint as it's usually available
+            target_datetime = datetime.now(tz.gettz(name=TZ)) - timedelta(minutes=15)
+
+        previous_quarter = _get_previous_quarter(target_datetime)
+
+        date = previous_quarter.strftime("%d-%b-%Y+%H:%M")
+
+        params = (
+            "?area="
+            + dataset
+            + "&region="
+            + region
+            + "&datefrom="
+            + date
+            + "&dateto="
+            + date
+        )
+
+        response: Response = session.get(DATA_URL + params)
+
+        return response.json()["Rows"][0]
+    except ConnectionError as e:
+        print("Failed to connect to SmartGrid Dashboard:" + e)
```

Comment:
```suggestion
        ParserException("GB_NIR.py", f"Failed to connect to SmartGrid Dashboard: {e}")
```
This uses ParserException instead of just printing the error.

This should be done for a few more later down as well.